File: benchmarks/javascript/json_roundtrip.js
const n = Number(process.argv[2] ?? 50000);
const records = [];
for (let i = 0; i < n; i++) {
    records.push({ id: i, name: "user" + i });
}
const text = JSON.stringify(records);
const back = JSON.parse(text);
let total = 0;
for (let i = 0; i < back.length; i++) {
    total = total + (back[i].id % 1000);
}
console.log(back.length, total, text.length);

File: benchmarks/ruby/json_roundtrip.rb
require "json"

n = (ARGV[0] || "50000").to_i
records = []
n.times { |i| records.push({ "id" => i, "name" => "user" + i.to_s }) }
text = JSON.generate(records)
back = JSON.parse(text)
total = 0
back.each { |record| total = total + record["id"] % 1000 }
puts "#{back.size} #{total} #{text.size}"
